pkg/apis/nsx.vmware.com/v1alpha1/subnet_types.go

```diff
@@ -78,20 +80,6 @@ type StaticIPAllocation struct {
 type DHCPConfig struct {
 	// +kubebuilder:default:=false
 	EnableDHCP bool `json:"enableDHCP,omitempty"`
```

Comment:
Why we remove all DHCP settings? I feel we should keep DHCPV4PoolSize and DNSClientConfig, and also add "ntp_servers".

BTW, what is the behavior when DHCP is enabled on the Subnet? Do we disable IPAM and MAC allocation for all Subnet Ports, or there is another flag on NetworkInterface to control DHCP or IPAM? @lxiaopei

Reply:
Since, as of now, we only use the EnableDHCP field in DHCPConfig. So, We think at this moment, we don't need to expose other properties. 
When DHCP is enabled on subnet, we will have the corresponding flag setting on the 
subnetports to control DHCP or IPAM.

https://github.com/vmware-tanzu/nsx-operator/blob/main/pkg/nsx/services/subnetport/builder.go#L36
https://github.com/vmware-tanzu/nsx-operator/blob/main/pkg/nsx/services/subnetport/builder.go#L70

@heypnus  please keep me honest. 

For DHCPV4PoolSize and DNSClientConfig, ntp_servers, we don't use it, do we still put them in CRD? @jwsui @heypnus

Reply:
I feel we need DNS and NTP settings in 9.0, and better to have DHCPV4PoolSize too. Could we add them?

Have we talked to VM Operator folks to make sure they will

Reply:
Sure, I will check with Qian for this part.

Reply:
There is another possibility - if we can expose VPCServiceProfile and make it modifiable in CCI apiserver, users can configure VPC level DHCP settings inc. DNS and NTP there, and then we can skip the settings on Subnet. Would let you decide which way to go.

Reply:
Sure. Another thing, I checked NSX API: DHCPV4PoolSize is not supported from NSX 4.2, so, we remove this field? 
For DNS and NTP settings in 9.0, we can see how to support it.

Reply:
Sure, we can remove deprecated fields in VPC 2.0.

For DNS/NTP, I am fine you want to remove them in this PR, and we can add them back after concluding the 9.0 plan.

Reply:
Sure, we can remove DHS/NTP, and will add back if necessary.